### PR TITLE
refactor

### PR DESCRIPTION
**Pull request by barnjamin**

### CODE REVIEW

**Comment by joe-p** [comment]
Looks like this adds spaces between void ops

```
store 1
store 2
store 3
```

```
store 1

store 2

store 3
```

I think ideally we keep sequential void ops together without spaces between

**Comment by barnjamin** [comment]
@joe-p pushed a change for the above

**Comment by joe-p** [comment]
Two other things I noticed:

1. If `#pragma` line already has a newline after it, an extra one gets added in making a double newline. 
2. Comments should have a newline above them to help readability

```c
#pragma version 7

test:
int 1
// two
int 2
// three
// four
int 3
int 4
```

Becomes

```c
#pragma version 7


test:

        int 1
        // two
        int 2
        // three
        // four
        int 3
        int 4
```

Which should ideally be

```
#pragma version 7

test:

        int 1

        // two
        int 2

        // three
        // four
        int 3
        int 4
```

File: tealfmt.go [hunk 1]
```diff
@@ -11,165 +11,158 @@ import (
...
-			return true
-		}
-	}
-	return false
-}
+	versionMatch = regexp.MustCompile(`^#pragma `)
+	opMatch      = regexp.MustCompile(`\S+`)
+	labelRegex   = regexp.MustCompile(`\S+:($| //)`)
+	commentRegex = regexp.MustCompile(`^//`)
+	voidOpMatch  = regexp.MustCompile("^(" + strings.Join(voidOps, "|") + ")$")
```
Comment (joe-p) [inline tealfmt.go:40]: The regex should end with `($| //)` so void ops followed by a comment are properly matched. 

I suppose to cover all cases it should really end with `($|\s*//)`

Comment (barnjamin) [reply tealfmt.go:40]: I pass the `op` in not the full line so I think this should still work?

**Pull request merged**